services/datamanager/datasync/sync.go

```diff
@@ -93,29 +97,41 @@ func NewManager(logger golog.Logger, partID string, client v1.DataSyncServiceCli
 ) (Manager, error) {
 	cancelCtx, cancelFunc := context.WithCancel(context.Background())
 	ret := syncer{
+		partID:             partID,
 		conn:               conn,
 		client:             client,
 		logger:             logger,
 		backgroundWorkers:  sync.WaitGroup{},
 		cancelCtx:          cancelCtx,
 		cancelFunc:         cancelFunc,
-		partID:             partID,
+		lastModifiedMillis: lastModifiedMillis,
 		progressLock:       &sync.Mutex{},
 		inProgress:         make(map[string]bool),
-		lastModifiedMillis: lastModifiedMillis,
+
+		syncErrs:   make(chan error, 10),
+		closed:     &atomic.Bool{},
+		logRoutine: sync.WaitGroup{},
 	}
+	ret.logRoutine.Add(1)
+	goutils.PanicCapturingGo(func() {
+		defer ret.logRoutine.Done()
+		ret.logSyncErrs()
+	})
 	return &ret, nil
 }
 
 // Close closes all resources (goroutines) associated with s.
 func (s *syncer) Close() {
+	s.closed.Store(true)
 	s.cancelFunc()
 	if s.conn != nil {
 		if err := s.conn.Close(); err != nil {
 			s.logger.Errorw("error closing datasync server connection", "error", err)
 		}
 	}
 	s.backgroundWorkers.Wait()
+	close(s.syncErrs)
+	s.logRoutine.Wait()
```

Comment:
Do we also want to  add `_ = s.logger.Sync()` here?

Reply:
Yup good call out! Added